c++/test/TestWriter.cc

```diff
@@ -209,5 +209,612 @@ namespace orc {
     }
     EXPECT_FALSE(rowReader->next(*batch));
   }
-}
 
+  TEST(Writer, writeStringAndBinaryColumn) {
```

Comment:
It is better to keep them in different test cases for following reasons:
1. It seems that they share the same logic in testing but they differ in the data construction and verification parts for different types. It has the same rationale with the design of ColumnWriter classes.
2. We may have different design/implementations for some types in the future. It is much easier to test them if we de-couple them into separate tests.